STYLE.md

```diff
@@ -0,0 +1,70 @@
+# Code Style Guide
+
+TLDR:
+- prioritise simple, straightforward code. Our users are researchers, often with little coding experience.
+- safety: use types, einops, jaxtyping, and liberal assertions.
+- fail fast - if something is wrong, the code should fail, not recover silently.
+
+
+## Design / Architecture
+
+We want to decouple metrics and analysis from the core codebase as much as possible, so that users can easily define their own and we don't need to make PRs to the codebase. See how this is done for core_metrics_and_figs.py.
+
+### Fail Fast (Negative Space Programming)
+Code should fail immediately when assumptions are violated, preventing bugs from propagating.
+
+If there's an assumption you're making while writing code, assert it.
+- If you were right, then it won't matter
+- If you were wrong, then the code **should** fail.
+
+```python
+# BAD - silently handles unexpected state
+def process_activations(acts):
+    if acts is None:
+        return torch.zeros(hidden_size)  # Hides the problem
```

Comment:
I can't see someone writing this function, but don't have an immediately better option. This might mean that providing examples here might not be worth it, and leaving just the bullet points above is fine. But if you have seen a lot of things like this, then I'm OK keeping it.